### PR TITLE
Storage Rule Add

### PR DESCRIPTION
**Pull request by Dala0**
Adds code to allow: -

1. Adding new storage rules.
2. Deletion of existing storage rules.

Tested on two virtual machines.

@fredex42

### CODE REVIEW

File: gnmvidispine/vs_storage_rule.py [hunk 1]
```diff
@@ -140,7 +140,12 @@ def add(self, newrule):
         if not isinstance(newrule,VSStorageRuleNew): raise TypeError("add() accepts only a VSStorageRuleNew object")
         newrule.assert_populated()
 
-
+        #pseudo-code
```
Comment (fredex42) [inline gnmvidispine/vs_storage_rule.py]: can remove this line!

File: tests/test_vsshape.py [hunk 2]
```diff
@@ -467,4 +469,24 @@ def test_download(self):
...
+        s = VSShape(host=self.fake_host, port=self.fake_port, user=self.fake_user, passwd=self.fake_passwd)
+        s.name = "VX-123"
+        s.contentDict = {'tag': 'original'}
+        s.itemid = "VX-456"
+        s.sendAuthorized = MagicMock(return_value=self.MockedResponse(200,self.test_storage_rule))
+        
+        s.add_storage_rule(newrule)
+        s.sendAuthorized.assert_called_with('PUT','/API/item/VX-456/storage-rule/original',
+                                            tostring(parsed_xml_doc),
+                                            {'Content-Type': 'application/xml', 'Accept': 'application/xml'})
```
Comment (fredex42) [inline tests/test_vsshape.py]: we also need a test for delete_storage_rule

File: gnmvidispine/vs_storage_rule.py (hunk 1 above)
Comment (Dala0) [inline gnmvidispine/vs_storage_rule.py]: Done.

File: tests/test_vsshape.py (hunk 2 above)
Comment (Dala0) [inline tests/test_vsshape.py]: I have built this.

**Pull request merged**